src/core_ocean/shared/mpas_ocn_mesh.F

```diff
@@ -851,30 +851,413 @@ end subroutine ocn_meshDestroy
+         edgeSignOnCellTmp,               &
+         edgeSignOnVertexTmp,             &
+         boundaryEdgeTmp,                 &
+         boundaryVertexTmp,               &
+         boundaryCellTmp
 
       !***
       !*** end of preamble, begin code
       !***
 
-#ifdef MPAS_OPENACC
-      !TODO: check for other fields that may need updating
       !$omp master
 
-      ! update fields on the device
-      !$acc update device(ocnMesh%edgeSignOnCell)           
+      ! already check during create that there should only be one block
+      block => domain % blocklist
+
+      ! retrieve the mpas mesh pool
+      call mpas_pool_get_subpool(block % structs, 'mesh', meshPool)
+
+      ! reset all mesh dimensions, just in case
+      call mpas_pool_get_dimension(meshPool, 'nCells', &
+                                              nCellsTmp)
+      call mpas_pool_get_dimension(meshPool, 'nEdges', &
+                                              nEdgesTmp)
+      call mpas_pool_get_dimension(meshPool, 'nVertices', &
+                                              nVerticesTmp)
+      call mpas_pool_get_dimension(meshPool, 'maxEdges', &
+                                              maxEdgesTmp)
+      call mpas_pool_get_dimension(meshPool, 'maxEdges2', &
+                                              maxEdges2Tmp)
+      call mpas_pool_get_dimension(meshPool, 'vertexDegree', &
+                                              vertexDegreeTmp)
+      call mpas_pool_get_dimension(meshPool, 'nVertLevels', &
+                                              nVertLevelsTmp)
+      call mpas_pool_get_dimension(meshPool, 'nVertLevelsP1', &
+                                              nVertLevelsP1Tmp)
+      call mpas_pool_get_dimension(meshPool, 'nAdvectionCells', &
+                                              nAdvectionCellsTmp)
+      call mpas_pool_get_dimension(meshPool, 'nCellsArray', &
+                                              nCellsArrayTmp)
+      call mpas_pool_get_dimension(meshPool, 'nEdgesArray', &
+                                              nEdgesArrayTmp)
+      call mpas_pool_get_dimension(meshPool, 'nVerticesArray', &
+                                              nVerticesArrayTmp)
+
+      ! translate scalar pointers to scalars in new mesh structure
+      ocnMesh%maxEdges        = maxEdgesTmp
+      ocnMesh%maxEdges2       = maxEdges2Tmp
+      ocnMesh%vertexDegree    = vertexDegreeTmp
+      ocnMesh%nVertLevels     = nVertLevelsTmp
+      ocnMesh%nVertLevelsP1   = nVertLevelsP1Tmp
+      ocnMesh%nAdvectionCells = nAdvectionCellsTmp
+
+      ! convert previous index limits into new arrangement
+      ocnMesh%nCellsAll       = nCellsTmp
+      ocnMesh%nEdgesAll       = nEdgesTmp
+      ocnMesh%nVerticesAll    = nVerticesTmp
+
+      n = size(nEdgesArrayTmp)
+      ocnMesh%nCellsOwned = nCellsArrayTmp(1)
+      do i=2,n
+         ocnMesh%nCellsHalo(i-1) = nCellsArrayTmp(i) 
+      end do
```

Comment:
After the most recent commit, when running the executable that was compiled with `DEBUG=true`, I get a segfault with the following traceback:
```
forrtl: severe (408): fort: (2): Subscript #1 of the array NCELLSARRAYTMP has value 5 which is greater than the upper bound of 4

Image              PC                Routine            Line        Source
ocean_model        0000000003332286  Unknown               Unknown  Unknown
ocean_model        0000000001EEB7BF  ocn_mesh_mp_ocn_m         963  mpas_ocn_mesh.F
ocean_model        0000000001B64939  ocn_forward_mode_         362  mpas_ocn_forward_mode.F
ocean_model        0000000001B5FB8B  ocn_core_mp_ocn_c          76  mpas_ocn_core.F
ocean_model        000000000041DE9F  mpas_subdriver_mp         331  mpas_subdriver.F
ocean_model        00000000004192E2  MAIN__                     14  mpas.F
ocean_model        0000000000419252  Unknown               Unknown  Unknown
libc-2.26.so       00002AAAADD4234A  __libc_start_main     Unknown  Unknown
ocean_model        000000000041916A  Unknown               Unknown  Unknown
```

I think line 960 should be `n = size(nCellsArrayTmp)` instead of `n = size(nEdgesArraytmp)`

Reply:
Making this change allowed me to run to completion with `DEBUG=true`

Reply:
Thanks. Yeah a cut-and-paste error. Fixed and pushed the change.